lib/SILOptimizer/Utils/InstOptUtils.cpp

```diff
@@ -1826,6 +1826,39 @@ void swift::endLifetimeAtLeakingBlocks(SILValue value,
       });
 }
 
+// Replace a store with an appropriate debug value.
+static void transferStoreDebugValue(DebugVarCarryingInst DefiningInst,
+                                    SILInstruction *SI,
+                                    SILValue original) {
+  auto VarInfo = DefiningInst.getVarInfo();
+  if (!VarInfo)
+    return;
+  if (!VarInfo->Loc && !SI->getLoc().hasSameSourceLocation(DefiningInst->getLoc()))
```

Comment:
Can you add a comment explaining at a high level what these conditions test for?